### PR TITLE
eos: Add metrics submission on parental controls blocking run/install

### PR DESCRIPTION
**Pull request by pwithnall**
UIs which present flatpaks should prevent the user from trying to run or
install apps which they aren’t allowed to because of their parental
controls policy.

Add metrics events from flatpak at the point where it prevents an app
being installed or run, so we have some statistics about possible places
where the UI is not sufficiently hiding disallowed flatpaks. All being
well, these metrics events should never be submitted.

This should not be upstreamed.

Signed-off-by: Philip Withnall <withnall@endlessm.com>

https://phabricator.endlessm.com/T28742

---

Debian changes in #211.

### CODE REVIEW

**Comment by pwithnall** [comment]
bot, retest this please

File: common/flatpak-dir.c [hunk 1]
```diff
@@ -7932,10 +7934,22 @@ flatpak_dir_check_parental_controls (FlatpakDir    *self,
...
 
   if (!authorized)
-    return flatpak_fail_error (error, FLATPAK_ERROR_PERMISSION_DENIED,
-                               /* Translators: The placeholder is for an app ref. */
-                               _("Installing %s is not allowed by the policy set by your administrator"),
-                               ref);
+    {
+      /* Send a metrics event so we have an idea if there are any places which
+       * don’t block app installing from the UI. If everything is implemented
+       * correctly, this event should never be submitted.
```
Comment (mwleeds) [inline common/flatpak-dir.c]: Maybe add "(unless the user initiated the install from the command line)" to this comment?

File: common/flatpak-dir.c [hunk 2]
```diff
@@ -7932,10 +7934,22 @@ flatpak_dir_check_parental_controls (FlatpakDir    *self,
...
+       *
+       * See: https://phabricator.endlessm.com/T28741 */
+#define FLATPAK_PARENTAL_CONTROLS_INSTALL_EVENT "9d03daad-f1ed-41a8-bc5a-6b532c075832"
```
Comment (mwleeds) [inline common/flatpak-dir.c]: This needs to be added to the wiki, and similarly for the `run` event.

File: common/flatpak-dir.c (hunk 1 above)
Comment (pwithnall) [reply common/flatpak-dir.c]: Good catch, done.

File: common/flatpak-dir.c (hunk 2 above)
Comment (pwithnall) [reply common/flatpak-dir.c]: Both done. https://phabricator.endlessm.com/phriction/diff/545/?l=41&r=42

**Pull request merged**